esphome/components/toshiba/toshiba.cpp

```diff
@@ -439,6 +766,227 @@ void ToshibaClimate::transmit_rac_pt1411hwru_temp_(const bool cs_state, const bo
   }
 }
 
+void ToshibaClimate::transmit_ras_2819t_() {
+  // Handle swing mode transmission for RAS-2819T
+  // Note: RAS-2819T uses a toggle command, so we need to track state changes
+
+  // Check if ONLY swing mode changed (and no other climate parameters)
+  bool swing_changed = (this->swing_mode != this->last_swing_mode_);
+  bool mode_changed = (this->mode != this->last_mode_);
+  bool fan_changed = (this->fan_mode != this->last_fan_mode_);
+  bool temp_changed = (abs(this->target_temperature - this->last_target_temperature_) > 0.1f);
+
+  bool only_swing_changed = swing_changed && !mode_changed && !fan_changed && !temp_changed;
+
+  if (only_swing_changed) {
+    ESP_LOGI(TAG, "[TX] Swing mode changed from %s to %s, sending ONLY toggle command",
+             this->last_swing_mode_ == climate::CLIMATE_SWING_VERTICAL ? "VERTICAL" : "OFF",
+             this->swing_mode == climate::CLIMATE_SWING_VERTICAL ? "VERTICAL" : "OFF");
```

Comment:
As above, save this for `verbose` level.
```suggestion
    ESP_LOGV(TAG, "[TX] Swing mode changed from %s to %s, sending ONLY toggle command",
             this->last_swing_mode_ == climate::CLIMATE_SWING_VERTICAL ? "VERTICAL" : "OFF",
             this->swing_mode == climate::CLIMATE_SWING_VERTICAL ? "VERTICAL" : "OFF");
```